crates/biome_js_analyze/src/lint/nursery/no_yoda_expression.rs

```diff
@@ -0,0 +1,416 @@
+                markup! {
+                    "Avoid the use of yoda expressions."
+                },
+            )
+            .note(markup! {
+                "Yoda expressions can be confusing to some people, invert the expression operands for better readability."
+            }),
+        )
+    }
+
+    fn action(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<JsRuleAction> {
+        let node = ctx.query();
+        let parent_statement = node.parent::<AnyJsStatement>();
+        let parent_expression = node.parent::<AnyJsExpression>();
+        let parent_yield_argument = node.parent::<JsYieldArgument>();
+        let mut mutation = node.clone().begin();
+
+        let left = node.left().ok()?;
+        let right = node.right().ok()?;
+        let operator_token = node.operator_token().ok()?;
+        let flipped_operator = flip_operator(node.operator().ok()?)?;
+
+        let left_leading_trivia = extract_leading_trivia(&left);
+        let left_trailing_trivia = extract_trailing_trivia(&left);
+        let right_leading_trivia = extract_leading_trivia(&right);
+        let right_trailing_trivia = extract_trailing_trivia(&right);
+        let operator_leading_trivia = operator_token.leading_trivia().pieces();
+        let operator_trailing_trivia = operator_token.trailing_trivia().pieces();
+        let whitespace = make::token(T!(==))
+            .with_trailing_trivia([(TriviaPieceKind::Whitespace, " ")])
+            .trailing_trivia()
+            .last();
+
+        let has_missing_left_trivia = match (&left_leading_trivia, &parent_yield_argument) {
+            (trivia, Some(parent_yield_argument)) => {
+                let parent_yield_expression = parent_yield_argument.parent::<JsYieldExpression>();
+                let has_trivia_on_parent_expression =
+                    parent_yield_expression.clone().is_some_and(|expression| {
+                        expression
+                            .yield_token()
+                            .is_ok_and(|yield_token| !yield_token.trailing_trivia().is_empty())
+                    });
+
+                trivia.is_empty() && !has_trivia_on_parent_expression
+            }
+            _ => false,
+        };
+        let has_missing_right_trivia = match (
+            right.clone().syntax().last_trailing_trivia(),
+            parent_statement.clone(),
+            parent_expression.clone(),
+        ) {
+            (_, Some(AnyJsStatement::JsIfStatement(_)), _) => false,
+            (_, Some(AnyJsStatement::JsWhileStatement(_)), _) => false,
+            (_, _, Some(AnyJsExpression::JsParenthesizedExpression(_))) => false,
+            (Some(trivia), _, _) => trivia.is_empty(),
+            (None, _, _) => true,
+        };
+
+        let new_left = if has_missing_left_trivia {
+            clone_with_trivia(right, &left_leading_trivia, &left_trailing_trivia)
+                .with_leading_trivia_pieces(whitespace.clone())
+                .unwrap()
```

Comment:
Don't unwrap

Reply:
done, removed all `.unwrap()` calls, but `.unwrap_or_default()` is ok, right?